src/main/java/ca/corefacility/bioinformatics/irida/ria/web/users/UsersAjaxController.java

```diff
@@ -84,6 +84,34 @@ public UsersAjaxController(UIUsersService UIUsersService) {
 
 	}
 
+	/**
+	 * Change the password for a user
+	 *
+	 * @param userId             The id of the user to edit (required)
+	 * @param oldPassword        The old password of the user for password change
+	 * @param newPassword        The new password of the user for password change
+	 * @param confirmNewPassword The confirmed new password of the user for password change
+	 * @param principal          a reference to the logged in user
+	 * @param request            the request
+	 * @return The name of the user view
+	 */
+	@RequestMapping(value = "/{userId}/changePassword", method = RequestMethod.POST)
+	public ResponseEntity<Map<String, String>> changeUserPassword(@PathVariable Long userId,
+			@RequestParam(required = false) String oldPassword, @RequestParam(required = false) String newPassword,
```

Comment:
These params should be `required=true` no?

Reply:
Please see 4680855c77b67c6c091779f2f8a7bfe0f0975be8.